rules/build_electricity.smk

```diff
@@ -20,7 +20,7 @@ if config["enable"].get("prepare_links_p_nom", False):
 
 rule build_electricity_demand:
     params:
-        snapshots={k: config["snapshots"][k] for k in ["start", "end", "inclusive"]},  # TODO: use config provider
+        snapshots={k: config_provider("snapshots", k) for k in ["start", "end", "inclusive"]},
```

Comment:
for such cases, the function comes at its limits. This is likely not working, as the return type of the config_provider("snapshots", k) is a function. Probably, we would have to go for something like 
```
    snapshots=lambda w: {k: config_provider("snapshots", k)(w) for k in ["start", "end", "inclusive"]},
```
It looks like there are some more cases like that in this commit.

To make it more readable, we can also rename `config_provider` to `configuration` or `conf`